src/jdk.jpackage/windows/native/common/Guid.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <algorithm>
+#include <Objbase.h>
+
+#include "Guid.h"
+#include "ErrorHandling.h"
+
+
+#pragma comment(lib, "ole32")
+
+
+Guid::Guid(const std::string& str) {
+    *this = Guid(std::wstring(str.begin(), str.end()));
+}
+
+
+namespace {
+
+void initGuid(const std::wstring& str, GUID& v) {
```

Comment:
Why it is separate method? Maybe make it static and part of class.

Reply:
This is internal helper method. My preference is to avoid them in class declaration to avoid visual noise.